pkg/statdb/statdb.go

```diff
@@ -200,14 +200,16 @@ func (s *Server) UpdateBatch(ctx context.Context, updateBatchReq *pb.UpdateBatch
 
 		updateRes, err := s.Update(ctx, updateReq)
 		if err != nil {
-			return nil, status.Errorf(codes.Internal, err.Error())
+			s.logger.Error(err.Error())
+			failedNodes = append(failedNodes, node)
+		} else {
+			nodeStatsList[i] = updateRes.Stats
```

Comment:
After changing how `nodeStatsList` is initialized, change this line to `nodeStatsList = append(nodeStatsList, updteRes.Stats)`